assignment_submission_checker/directory.py

```diff
@@ -406,8 +404,18 @@ def check_files(self, directory: Path) -> Tuple[Set[str], Set[str], Set[str]]:
             file for pattern in self.data_file_patterns for file in fnmatch.filter(files, pattern)
         )
         unexpected = files - set(self.compulsory) - set(self.optional) - data_files
+        if self.git_root:
+            # Do not report git files as unexpected if we're at the git root.
+            # Add these as optionals
+            git_files = set()
+
+            for file in unexpected:
+                if any(fnmatch(file.lower(), pattern.lower()) for pattern in GIT_ROOT_PATTERNS):
+                    git_files.add(file)
+
+            unexpected = unexpected - git_files
 
-        optional = files - unexpected - set(self.compulsory)
+        optional = files - unexpected - set(self.compulsory) + git_files
```

Comment:
```suggestion
        optional = (files - unexpected - set(self.compulsory)).union(git_files)
```

Sets don't have an + overload, Will.